### ISSUE TITLE
Change left navigation menu title with a simple variable

### ISSUE THREAD

**Issue by EdouardThiel** [squidfunk/mkdocs-material#2550]
By default, the site name is used as the title for the left navigation menu.

This is not always desirable; for instance if the site name is long, it could be replaced by a short title or simply by "Navigation".

For now, material does not offer a simple solution, and I would like to suggest one:

1) In `material/partial/nav.html`, replace line 16

        {{ config.site_name }}

    with

        {% if nav_title|trim %}{{ nav_title }}{% else %}{{ config.site_name }}{% endif %}

2) The user will then, if he likes to, add in `overrides/main.html`:

        
        {% set nav_title %}My navigation title{% endset %}

I have used `nav_title` because this variable is unused in Material, but another name could fit too, e.g. `extranavtitle`.

Regarding the documentation, the natural place would be in Getting started / customization, in the last table with `extracopyright`.

**Comment by squidfunk** [comment]
Thanks for your suggestion. I'm not sure if this is the right way to implement this behavior. Note that when you set the navigation title like you propose in the linked PRs (thanks for those, by the way), you will also override the title in the drawer which, in my opinion, should definitely show the site title. This is because the sidebar is re-used on mobile (in order to minimize duplicated content) and rendered as a drawer which collapses into the side:

<img width="242" alt="Bildschirmfoto 2021-04-13 um 18 27 23" src="https://user-images.githubusercontent.com/932156/114587406-ed415f80-9c85-11eb-89c0-9eac502699da.png">

Thus, I'm not considering including this feature request / behavior. After all, overriding `nav.html` is fairly easy and the navigation partial didn't change much during the last major updates (last time a year ago with 5.x).

**Comment by EdouardThiel** [comment]
Well this allows short titles, even on mobiles, not all sites have a short title as "material for mkdocs" and this features is actually missing, there are questions about it on stack overflow.This is a simple solution. Another would be to define config.short_nav_title in config. May I submit it?-- Professeur Edouard THIEL                  http://j.mp/ethiel
Département Informatique et Interactions  http://j.mp/dept-ii
UFR Sciences, Aix-Marseille Université    http://j.mp/fs-amu
Laboratoire d'Informatique et Systèmes    http://j.mp/lis-amu

-------- Message d'origine --------De : Martin Donath ***@***.***> Date : 13/04/2021  18:28  (GMT+01:00) À : squidfunk/mkdocs-material ***@***.***> Cc : EdouardThiel ***@***.***>, Author ***@***.***> Objet : Re: [squidfunk/mkdocs-material] Change left navigation menu title with a simple variable (#2550) 
Thanks for your suggestion. I'm not sure if this is the right way to implement this behavior. Note that when you set the navigation title like you propose in the linked PRs (thanks for those, by the way), you will also override the title in the drawer which, in my opinion, should definitely show the site title. This is because the sidebar is re-used on mobile (in order to minimize duplicated content) and rendered as a drawer which collapses into the side:

Thus, I'm not considering including this feature request / behavior. After all, overriding nav.html is fairly easy and the navigation partial didn't change much during the last major updates (last time a year ago with 5.x).

**Comment by EdouardThiel** [comment]
Well my proposal is to allow nav short titles, even on mobiles: not all sites have a short title as "Material for Mkdocs" and this features is actually missing (there are questions about it on Stack Overflow and other places).

Suppose the site title is "Course of System and Network Administration by Alec Tricity".

Without the ability of nav short title we'll get on a large screen:

![screen1](https://user-images.githubusercontent.com/61007915/114852819-6dc9a280-9de3-11eb-8609-33af9e0de593.png)

And on a mobile screen:

![screen2](https://user-images.githubusercontent.com/61007915/114852825-6efacf80-9de3-11eb-9fa5-5f37577ee80e.png)

After clicking on the menu icon we get:

![screen3](https://user-images.githubusercontent.com/61007915/114852828-6f936600-9de3-11eb-915c-0f34ce7af97b.png)

The title is seen three times in the first figure (windows title, bar title and nav title)! Here is what we can get with my proposal, with `nav_title = "Navigation"` :

![screen4](https://user-images.githubusercontent.com/61007915/114852838-71f5c000-9de3-11eb-8fd7-2a8663aac665.png)

On a mobile, when the user clicks on the menu, he perfectly see the full title (figure 2) and there is no need to print again the full title in the drop down menu (default situation). Here is what we get with my proposal (and still `nav_title = "Navigation"`, but replace with what you like):

![screen5](https://user-images.githubusercontent.com/61007915/114852842-728e5680-9de3-11eb-8dc3-7e525193e845.png)

This proposal does not destroy the look and feel of Material, but enhances it.
The proposed solution is very simple with a variable `nav_title`. Another would be to define `config.short_nav_title` in `mkdocs.yml`.

Overriding `nav.html` is not a good solution because if a user updates its version of Material, maybe the overridden `nav.html` could become inconsistent with the new Material code.

**Comment by squidfunk** [comment]
Thanks for coming back to this topic. The thing is: I'm not interested in providing more and more template variables for things like that. There are now two ways to define overrides: blocks and variables, and I'm not sure if this is a good idea. Yes, we added `extracopyright`, but I'm still trying to figure out whether this was a good decision. I really don't want to introduce more and more complexity, because, in the end, it will lead to more confusion and more issues which I have to answer (because of the complexity), taking away my time to implement new features and fix bugs. Currently, there are four potential places, which we use to source data to the template:

- `mkdocs.yml`
- `partials/languages/*.html` (translations)
- template blocks via overrides
- template variables via overrides

That's too many. Please remember: this is an opinionated project. I encourage you to read https://github.com/squidfunk/mkdocs-material/discussions/2399.

**Comment by EdouardThiel** [comment]
I understand, although I think that adding some few hooks can be useful and doesn't add extra complexity or bug risks.

By the way I would like to thank you for this project which is amazing. I have tried a number of site generators, and write some, for my own site, and I wanted to find one for my own courses, which allows: markdown, mobile and search box, with beautiful rendering: for these goals Material is perfect.

On the other hand, one important thing was missing for me: to publish courses by filtering them (from course 1 to course n) to make them appear gradually. That's why I started to write a script to add this feature, and others like creating a virtualenv, etc, and finally I published it for colleagues: it is named [Mkdocs-et](http://pageperso.lif.univ-mrs.fr/~edouard.thiel/mkdocs-et/)

Since my script generates its own virtualenv, it now also directly updates material/partials/nav.html and add the nav_title variable in it, defined as "Navigation" in overrides/main.html. Since I will use it for several courses, each with its own config directory (skel-mkdocs-et/), and one different version each year, each time I will update the venv, the original nav.html will be modified and available for each course/year in my directories.
